pat-static/src/main/java/com/rectanglel/patstatic/model/PatApiServiceImpl.kt

```diff
@@ -0,0 +1,169 @@
+import com.rectanglel.patstatic.predictions.response.PredictionResponse
+import com.rectanglel.patstatic.routes.BusRoute
+import com.rectanglel.patstatic.routes.RoutesDataManager
+import com.rectanglel.patstatic.utils.Constants
+import com.rectanglel.patstatic.vehicles.response.VehicleResponse
+import com.rectanglel.patstatic.wrappers.WifiChecker
+import io.reactivex.Flowable
+import io.reactivex.Single
+import io.reactivex.schedulers.Schedulers
+import okhttp3.OkHttpClient
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.gson.GsonConverterFactory
+import java.io.File
+import java.text.SimpleDateFormat
+import java.util.*
+import java.util.concurrent.TimeUnit
+
+/**
+ * Implemented service that retrieves data.
+ * Created by epicstar on 9/18/16.
+ * @author Jeremy Jao
+ * @since 78
+ */
+class PatApiServiceImpl(baseUrl: String,
+                        apiKey: String,
+                        dataDirectory: File,
+                        staticData: StaticData,
+                        wifiChecker: WifiChecker) : PatApiService {
+
+    private val patApiClient: RetrofitPatApi
+
+    private val patternDataManager: PatternDataManager
+
+    private val routesDataManager: RoutesDataManager
+
+    override val routes: Single<List<BusRoute>>
+        get() = routesDataManager.routes
+
+    init {
+
+        patApiClient = createPatApiClient(baseUrl, apiKey)
+        patternDataManager = PatternDataManager(dataDirectory, patApiClient, staticData, wifiChecker)
+        routesDataManager = RoutesDataManager(dataDirectory, patApiClient, staticData)
+    }
+
+    override fun getPatterns(rt: String): Flowable<List<Ptr>> {
+        return patternDataManager.getPatterns(rt)
+                .compose(applySchedulers())
+    }
+
+    override fun getStops(rt: String): Flowable<List<Pt>> {
+        throw NotImplementedError()
+    }
+
+    override fun getVehicles(routes: Collection<String>): Flowable<VehicleResponse> {
+        return patApiClient.getVehicles(routes.joinToString(","))
+                .compose(applySchedulers())
+    }
+
+    override fun getStopPredictions(stpid: Int, rts: Collection<String>): Single<List<Prd>> {
+        return patApiClient.getStopPredictions(stpid, rts.joinToString(","))
+                .compose(composePrds())
+    }
+
+    override fun getVehiclePredictions(vid: Int): Single<List<Prd>> {
+        return patApiClient.getBusPredictions(vid)
+                .compose(composePrds())
+
+    }
+
+    companion object {
+        /**
+         * This is the date format to parse
+         *
+         * @since 46
+         */
+        private const val DATE_FORMAT_PARSE = "yyyyMMdd HH:mm"
+
+        /**
+         * The default date format to parse... The timezone is set as EST
+         *
+         * @since 46
+         */
+        private val DEFAULT_DATE_PARSE_FORMAT = SimpleDateFormat(DATE_FORMAT_PARSE, Locale.US)
```

Comment:
soon we should start using threeten abp: https://github.com/JakeWharton/ThreeTenABP

but don't change until later

Reply:
I'll make a ticket

Reply:
wait it looks like it's deprecated lmao